src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -699,7 +699,7 @@ private BoundDecisionDag MakeBoundDecisionDag(SyntaxNode syntax, ImmutableArray<
 
             var rootDecisionDagNode = decisionDag.RootNode.Dag;
             RoslynDebug.Assert(rootDecisionDagNode != null);
-            var boundDecisionDag = new BoundDecisionDag(rootDecisionDagNode.Syntax, rootDecisionDagNode);
+            var boundDecisionDag = new BoundDecisionDag(rootDecisionDagNode.Syntax, rootDecisionDagNode).RemoveCompilerGeneratedNodes();
```

Comment:
nit: I'm debating whether using the "compiler generated" flag is better than introducing a dedicated flag (maybe "subsumption only" or some other name).